### ISSUE TITLE
Want to ask something about StyleRig

### ISSUE THREAD

**Issue by johannwyh** [yukyeongleee/pggan#1]
Hello, dear authors!

I find that you are working on reproducing StyleRig. Have you ever trained its **DFR** model? Have you succeeded on it and how is it performing? 

Thank you so much for your help.

BTW, if you have anything to ask about HifiFace, feel free to contact me LOL.

**Comment by 1zong2** [comment]
Hi, thank you for your interest.

Yes, we have trained the DFR model before, but we did not used it to implement StyleRig.
Because it is easier and more accurate to get p using a [3d reconstruction model](https://github.com/sicxu/Deep3DFaceRecon_pytorch) 

So our approach is not the same as the original paper. 
original paper: v(18X512) -> DFR model -> p(257)
our approach: v(18X512) -> stylegan2 generator -> Image (1024) -> 3d face recon model -> p(257)

Moreover, we set a loss function of the generator as below, which is far from the methods in the original paper. 
_Total loss = 10 * L_3d_reconstructed_face + 10 * L_3d_landmark + 1 * L_id_from_arcface + L_adv/18_
**(We are inspired by your beautiful face swapping model.)**

Also, we used a [LatentCodesDiscriminator](https://github.com/omertov/encoder4editing/blob/6f8d68a9fed7dad3b5f1ea1b65835a28d8f6fd97/models/discriminator.py) from e4e model.

We shared some [results](https://drive.google.com/file/d/13UDHKZ_jlqlLdhFvzp_0qxS7wMxWY7TY/view?usp=sharing
) of our implementation.
You can see there are some problems such as a large pose transition error, entanglements of attributes, out-of-domain latent vectors, etc.

**Comment by 1zong2** [comment]
@johannwyh  

I have read all the issues in your HIFIFACE repository, and thanks for the kind explanation.

I also have some questions.

1. Purpose of L_seg
The effect of L_seg is shown in Figure 9 of your paper.
But I think that M_low and M_r will eventually have same appearance with M_tar because lambda of L_seg is much higher than others.
So I trained without L_seg and replaced the M_low and the M_r with the M_tar.
Comparing the two options with and without L_seg, I found that elements such as eyes, nose, and mouth are slightly more distinct in z_dec.
Is the functionality of L_seg to make z_dec more focused on the eyes, nose and mouth?
If M_low has the same shape as M_tar, I wonder what the role of L_seg will be.

2. F_up module
Maybe Res-blocks in F_up module have 256 channels.
Did you use 1x1 conv block to make 3-channel image? or the last Res-block has input dim of 256 and output dim of 3?
It is difficult to determine whether it is better to use a 1x1 conv layer in order to reduce the channels.

3. Video-level face swapping test
Did you apply your model to videos?
I wonder if the face shape change will be continuously and naturally maintained.
